.github/workflows/tests.yml

```diff
@@ -73,3 +73,10 @@ jobs:
         with:
           file: ./coverage.txt
         if: env.GIT_DIFF
+  
+  test_persistence:
+    runs-on: ubuntu-latest
+    steps:
+      - name: run persistence tests
+        run: ./test/persist/test_failure_indices.sh
+        shell: bash
```

Comment:
I believe that this line controls which shel the "run" line is run in, and you shouldn't need to wrap it in an extra bash session, which will only cause confusion